rfc/rfc017-distributed-network-grpc-v2.md

```diff
@@ -45,4 +45,20 @@ As specified by RFC015, the node MUST authenticate the peer's node DID as follow
 1. Resolve peer's node DID to its corresponding DID document.
 2. Assert that one of the `NutsComm` endpoint's host matches (one of) the `dNSName` SANs in the peer's TLS client certificate.
 
-## X. Private Transactions
\ No newline at end of file
+## X. Private Transactions
+
+When the node receives a transaction that contains a `pal` header, the transaction is considered private.
+This means the transaction contents can only be retrieved by (and should only be shared with) the participants listed in the `pal` header.
+Since the `pal` header is encrypted (see [RFC004](rfc004-verifiable-transactional-graph.md)) to preserve anonymity of the participants,
+it must be decrypted first using the node's `keyAgreement` keys from its DID document.
+If the local node can decrypt the `pal` header, it means the transaction is (also) intended for the local node.
```

Comment:
>How does this work with key rotation. What key should be used to decrypt?

All decryption keys should be tried. But since you know when you published a new encryption key (not in terms of time, but in DAG TX ordering) you know from what point on you can expect private TXs with your DID in the encrypted `pal` header to be published. Vice versa, you can remove an encryption key and you can assume no new TXs are published, encrypted with your old encryption key after those peers have processed your DID document update.

> Is it allowed to just request the payload from the creator of the Tx to see if I'm in one of the recipients?

That would work, but you can't derive much information from it, since encrypted PAL header contains a random seed to mitigate correlation attacks. It would also be noticable, since when a peer requests a private TX while not being in the `pal` header, you can be sure it's an attempted attack (or a bug in their software). So monitorable and punishable.